### ISSUE TITLE
bug in when using there is mismatch between number of labels in training and testing data set.

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/multilevelmod#38]
I found this problem in https://stackoverflow.com/questions/72349462/r-using-a-lmer-model-in-fit-resamples-fails-with-error-assigned-data-facto and condensed it further down, to this:

@hfrick:
> that seems weird. multilevelmod::reformat_lme_pred_data() breaks because there are no factor levels. I quickly checked if it makes a difference if you turn manufacturer and model into factors before you do anything else but that’s not it.

```r
library(tidyverse)
library(tidymodels)
library(multilevelmod)

data(mpg, package = "ggplot2")

lmm_model = linear_reg() %>% 
  set_engine("lmer")

lmm_workflow = workflow() %>% 
  add_variables(outcomes = cty,
                predictors = c(year, manufacturer, model)) %>% 
  add_model(lmm_model, formula = cty ~ year + (1|manufacturer/model))

# A simple fit works
lmm_fit <- fit(lmm_workflow, mpg)

predict(lmm_fit, new_data = mpg)
#> Error:
#> ! Assigned data `factor(lvl[1], levels = lvl)` must be compatible with existing data.
#> ✖ Existing data has 234 rows.
#> ✖ Assigned data has 0 rows.
#> ℹ Only vectors of size 1 are recycled.
```

**Comment by a-difabio** [comment]
It looks like to me, that the only way to make an lmer model predict on new factor levels is to specify `allow.new.levels = TRUE` in the `predict()` call. However, this is not possible when calling `predict()` on a Workflow object.

```
# This also works; see ?lme4::predict.merMod
predict(lmm_fit %>% extract_fit_engine(), newdata = mpg, allow.new.levels = TRUE)

predict(lmm_fit, new_data = mpg, allow.new.levels = TRUE)
#> Error: The ellipses are not used to pass args to the model function's predict function. These arguments cannot be used: `allow.new.levels`
```

**Comment by hfrick** [comment]
closed in #41
